--- feeds/luci/luci/Makefile	2015-01-22 08:10:27.600768118 +0100
+++ feeds/luci/luci//Makefile	2015-01-22 19:13:57.602733682 +0100
@@ -14,7 +14,7 @@ PKG_BUILD_PARALLEL:=0
 ifeq ($(USELOCAL),1)
   PKG_VERSION:=0.11+svn
 else
-  PKG_SOURCE_URL:=http://svn.luci.subsignal.org/luci/$(PKG_BRANCH)
+  PKG_SOURCE_URL:=https://github.com/openwrt/luci/$(PKG_BRANCH)
   ifeq ($(DUMP),)
     PKG_REV:=$(shell LC_ALL=C svn info $(CURDIR) | sed -ne's/^Revision: //p')
     PKG_VERSION:=0.11+svn$(PKG_REV)
